src/tiled/objecttypesmodel.cpp

```diff
@@ -141,4 +158,5 @@ void ObjectTypesModel::appendNewObjectType()
     beginInsertRows(QModelIndex(), mObjectTypes.size(), mObjectTypes.size());
     mObjectTypes.append(ObjectType());
     endInsertRows();
+    sort(0, Qt::AscendingOrder);
```

Comment:
The new object type will have an empty name. Rather than sorting afterwards, we could probably just prepend instead of append?

`ObjectTypesModel::appendNewObjectType` should be renamed to `ObjectTypesModel::addNewObjectType`, and I think it should return a `QModelIndex` pointing to the added object type (which can be assigned to `newIndex` in `ObjectTypesEditor::addObjectType`, to avoid making assumptions about where the new object is added there).

Reply:
Great suggestion. I wasn't happy with this setup either. That's a good way to clean it up!